TombEngine/Objects/TR4/Entity/tr4_baddy.cpp

```diff
@@ -1312,7 +1313,30 @@ namespace TEN::Entities::TR4
 			item->Animation.ActiveState = BADDY_STATE_BLIND;
 			creature->MaxTurn = 0;
 		}
+	}
 
-		return;
+	void Baddy2Hit(ItemInfo* itemhit, ItemInfo* insticator, std::optional<GameVector> hitPos, int damage, int grenade, short meshHit)
+	{
+		const auto& lara = *GetLaraInfo(insticator);
+		const auto& object = Objects[itemhit->ObjectNumber];
+		if (object.hitEffect != HitEffect::None && hitPos.has_value())
+		{
+			switch (object.hitEffect)
```

Comment:
Why a switch with just a single case? If it's a prototype for block state (when he swirls a sword in front of himself), shouldn't a state be checked instead of `object.hitEffect`?

Reply:
copy-paste will fix it.

Reply:
Done